src/enable_variable_ops/ngraph_catalog.cc

```diff
@@ -111,5 +113,53 @@ bool NGraphCatalog::ExistsInInputVariableSharedNameMap(int graphid,
       NGraphCatalog::CreateNodeKey(graphid, node_name, input_index));
 }
 
+// Functions for EncapOutputInfo Map
+void NGraphCatalog::AddToEncapOutputInfoMap(string key,
+                                            tuple<string, bool, bool> val) {
+  NGraphCatalog::encap_output_info_map_[key] = val;
+}
+
+void NGraphCatalog::AddToEncapOutputInfoMap(string key, string shared_name,
+                                            bool copy_to_tf,
+                                            bool is_tf_just_looking) {
+  // create a tuple
+  tuple<string, bool, bool> val =
+      make_tuple(shared_name, copy_to_tf, is_tf_just_looking);
+  NGraphCatalog::encap_output_info_map_[key] = val;
+}
+
+bool NGraphCatalog::ExistsInEncapOutputInfoMap(string key) {
+  auto itr = NGraphCatalog::encap_output_info_map_.find(key);
+  return itr != NGraphCatalog::encap_output_info_map_.end();
+}
+
+bool NGraphCatalog::ExistsInEncapOutputInfoMap(int graphid, string node_name,
+                                               int input_index) {
+  std::string key =
+      NGraphCatalog::CreateNodeKey(graphid, node_name, input_index);
+  auto itr = NGraphCatalog::encap_output_info_map_.find(key);
+  return itr != NGraphCatalog::encap_output_info_map_.end();
+}
+
+tuple<string, bool, bool> NGraphCatalog::GetInfoFromEncapOutputInfoMap(
+    string key) {
+  return NGraphCatalog::encap_output_info_map_[key];
```

Comment:
optional: we can change all the Get api's to use map.at(key) instead of map[key] as they expect the key to be present